### PR TITLE
Adding documentation for default value can not be overridden in the parent entity if value is provided by the child entity

### PR DESCRIPTION
**Pull request by brijeshdhakar**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the CODEOWNERS. 

## What Type of PR is This: This PR adds the documentation that a parent entity can not override the default value for any input if the default value is defined by child entity.

Screenshot 2024-03-14 at 4 49 47 PM

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screenshot.

### CODE REVIEW

File: docs/platform/variables-and-expressions/runtime-inputs.md [hunk 1]
```diff
@@ -212,6 +212,10 @@ pipeline:
                 value: <+input>.default(new york).executionInput()
 ```
 
+If you want to provide the default value in the parent entity(Pipeline), then you can define the field value as `<+input>` in the child entity(Template). And in parent entity(Pipeline) you can define the default value like `<+input>.deault(default_val)` for any input.
```
Comment (SudarshanPurohit) [inline docs/platform/variables-and-expressions/runtime-inputs.md:215]: (This is a little hard to follow. Let's break it into more detail: - Sudarshan)

For any variable defined in a template, the `default` method can be added to either its declaration in the child entity(Template in this case), or else in the parent entity (Pipeline), but not both. Here's what the two different usages mean: 

* When the default value is specified in the child entity, all other entities that use it, cannot specify a different default. They can, however, specify a different value directly, by writing a fixed value for the variable. 
* When there's no default value specified in the child entity, the parent entities may or may not add a default, and this default value can be different in each of these parents.

Comment (SudarshanPurohit) [reply docs/platform/variables-and-expressions/runtime-inputs.md:215]: One more thing to check - if there's a default value specified in the Template, can we remove the reference to that variable completely in the YAML? Just check, and we can add that into the second paragraph.

Comment (brijeshdhakar) [reply docs/platform/variables-and-expressions/runtime-inputs.md]: Addressed the comments.

**Review by brian-f-harness — APPROVED** [review]

**Review by SudarshanPurohit — DISMISSED** [review]

File: docs/platform/variables-and-expressions/runtime-inputs.md [hunk 2]
```diff
@@ -212,6 +212,14 @@ pipeline:
                 value: <+input>.default(new york).executionInput()
 ```
 
+For any variable defined in a template, the default method can be added to either its declaration in the child entity(Template in this case), or else in the parent entity (Pipeline), but not both. Here's what the two different usages mean:
```
Comment (brian-f-harness) [inline docs/platform/variables-and-expressions/runtime-inputs.md:215]:
```suggestion
For any variable defined in a template, the default method can be added to either its declaration in the child entity (Template in this case), or else in the parent entity (Pipeline), but not both. Here's what the two different usages mean:
```

File: docs/platform/variables-and-expressions/runtime-inputs.md [hunk 3]
```diff
@@ -212,6 +212,14 @@ pipeline:
...
+- When the default value is specified in the child entity, all other entities that use it, cannot specify a different default. They can, however, specify a different value directly, by writing a fixed value for the variable.
```
Comment (brian-f-harness) [inline docs/platform/variables-and-expressions/runtime-inputs.md:216]:
```suggestion
- When the default value is specified in the child entity, all other entities that use it cannot specify a different default. They can, however, specify a different value directly, by writing a fixed value for the variable.
```

File: docs/platform/variables-and-expressions/runtime-inputs.md [hunk 4]
```diff
@@ -212,6 +212,14 @@ pipeline:
...
+
+  For example if you have specified any input like `<+input>.default(default_val)` in child entity then you can not override this default in the parent entity.
```
Comment (brian-f-harness) [inline docs/platform/variables-and-expressions/runtime-inputs.md:218]:
```suggestion
  For example, if you have specified any input like `<+input>.default(default_val)` in the child entity, then you cannot override this default in the parent entity.
```

File: docs/platform/variables-and-expressions/runtime-inputs.md [hunk 5]
```diff
@@ -212,6 +212,14 @@ pipeline:
...
+- When there's no default value specified in the child entity, the parent entities may or may not add a default, and this default value can be different in each of these parents.
+
+  For example if you have specified input like `<+input>` for any field then you can provide default in the parent entity like `<+input>.default(default_val_in_parent)`.
```
Comment (brian-f-harness) [inline docs/platform/variables-and-expressions/runtime-inputs.md:221]:
```suggestion
  For example, if you have specified input like `<+input>` for any field, then you can provide a default in the parent entity like `<+input>.default(default_val_in_parent)`.
```

**Review by brian-f-harness — APPROVED** [review]
